app/src/main/java/com/github/jorgecastillo/kotlinandroid/data/HeroesRepository.kt

```diff
@@ -2,7 +2,11 @@ package com.github.jorgecastillo.kotlinandroid.data
 
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchAllHeroes
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchHeroesFromAvengerComics
+import com.karumi.marvelapiclient.model.CharacterDto
+import kategory.HK
 
-fun getHeroesWithCachePolicy() = fetchAllHeroes()
+inline fun <reified F> getHeroesWithCachePolicy(): HK<F, List<CharacterDto>> =
```

Comment:
You can get rid of all the inline stuff if you just use `AsyncResult` directly instead of `MonadControl` and then you don't need to reify anything, but that would couple your impl to an async runtime and the purpose of tagless final is that you can separate declaration from interpretation